src/codemodder/project_analysis/file_parsers/base_parser.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+from pathlib import Path
+from typing import List
+from .package_store import PackageStore
+from packaging.requirements import Requirement
+
+
+class BaseParser(ABC):
+    def __init__(self, parent_directory: Path):
+        self.parent_directory = parent_directory
+
+    @property
+    @abstractmethod
+    def file_name(self):
+        ...  # pragma: no cover
```

Comment:
Just a tiny thing but I think you can have the `...` on the same line as the `def` which probably makes the `# pragma` unnecessary.

Reply:
oh I hope so!

Reply:
black doesn't like inline ... so I left it as is